crates/bevy_asset/src/io/file/mod.rs

```diff
@@ -0,0 +1,323 @@
+#[cfg(feature = "filesystem_watcher")]
+mod file_watcher;
+
+use crate::io::{
+    get_meta_path, AssetReader, AssetReaderError, AssetWatcher, AssetWriter, AssetWriterError,
+    PathStream, Reader, Writer,
+};
+use anyhow::Result;
+use async_fs::{read_dir, File};
+use bevy_utils::BoxedFuture;
+use futures_lite::StreamExt;
+
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+pub(crate) fn get_base_path() -> PathBuf {
+    if let Ok(manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
```

Comment:
we might want to keep another additional variable for a better control over base_path:
behaviour on master: https://github.com/bevyengine/bevy/pull/8624/files#diff-b659da4ebc9eec5698cafa0b10611f602d3821ff68f664f8c88f259275d37b5cL63-L72)

context for this change: https://github.com/bevyengine/bevy/pull/5346

Reply:
Good call. (Re)added!